vpr/src/util/vpr_utils.cpp

```diff
@@ -578,21 +618,33 @@ int find_pb_pin_clb_pin(ClusterBlockId clb, int pb_pin) {
     auto& cluster_ctx = g_vpr_ctx.clustering();
     auto& place_ctx = g_vpr_ctx.placement();
 
-    int clb_pin = -1;
+    auto& clb_nlist = cluster_ctx.clb_nlist;
+
+    int eq_type_index = clb_nlist.block_eq_type_index(clb);
+    auto type = clb_nlist.block_type(clb);
+
+    int pin = pb_pin;
+
+    // In case an equivalent tile is selected, the CLB block type will be different (e.g. CLB logic type is LAB, CLB physical type is MLAB).
+    // Therefore, I need to retrieve the pin mapping from the LAB type by setting the `false` flag when calling block_type.
+    if(eq_type_index != OPEN) {
+        pin = clb_nlist.block_type(clb, false)->equivalent_tile_pin_mapping[eq_type_index][pb_pin];
```

Comment:
Rather than this logic, can we add "block_type_effective" or something?  That internalizes the "block_eq_type_index"?  This method dispatch seems redundant.

Reply:
Given that this pattern has appeared twice in one file, this change must happen.

Reply:
You mean internalizing the check to see whether the block has been placed in an equivalent tile or not, right? Sounds good to me.

Reply:
Ya, there is not reason to leave it up to the caller.

Reply:
Done